lib/ts/recipe/thirdparty/index.ts

```diff
@@ -0,0 +1,143 @@
+import Recipe from "./recipe";
+import { InputType, StateObject, PreAndPostAPIHookAction, PreAPIHookContext, PostAPIHookContext } from "./types";
+
+export default class Wrapper {
+    static init(config: InputType) {
+        return Recipe.init(config);
+    }
+
+    static getOAuthState(input?: { userContext?: any }): {
+        status: "OK";
+        state: StateObject | undefined;
+    } {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.getOAuthState({
+            userContext: getNormalisedUserContext(input?.userContext),
+            config: recipeInstance.config,
+        });
+    }
+
+    static setOAuthState(input: { state: StateObject; userContext?: any }): {
+        status: "OK";
+    } {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.setOAuthState({
+            ...input,
+            userContext: getNormalisedUserContext(input.userContext),
+            config: recipeInstance.config,
+        });
+    }
+
+    static getThirdPartyLoginRedirectURLWithQueryParams(input: {
+        thirdPartyProviderId: string;
+        thirdPartyRedirectionURL: string;
+        state?: StateObject;
+        userContext?: any;
+    }): Promise<
+        | {
+              status: "ERROR";
+          }
+        | {
+              status: "OK";
+              url: string;
+          }
+    > {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.getThirdPartyLoginRedirectURLWithQueryParams({
+            ...input,
+            config: recipeInstance.config,
+            userContext: getNormalisedUserContext(input.userContext),
+        });
+    }
+
+    static getOAuthAuthorisationURL(input: {
+        thirdPartyProviderId: string;
+        userContext?: any;
+        options?: RecipeFunctionOptions;
+    }): Promise<{
+        status: "OK";
+        url: string;
+        fetchResponse: Response;
+    }> {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.getOAuthAuthorisationURL({
+            ...input,
+            config: recipeInstance.config,
+            userContext: getNormalisedUserContext(input.userContext),
+        });
+    }
+
+    static signInAndUp(input: {
+        thirdPartyProviderId: string;
+        thirdPartyRedirectionURL: string;
+        thirdPartyProviderClientId?: string;
+        userContext?: any;
+        options?: RecipeFunctionOptions;
+    }): Promise<
+        | {
+              status: "OK";
+              user: UserType;
+              createdNewUser: boolean;
+              fetchResponse: Response;
+          }
+        | {
+              status: "NO_EMAIL_GIVEN_BY_PROVIDER";
+              fetchResponse: Response;
+          }
+        | {
+              status: "FIELD_ERROR";
+              error: string;
+              fetchResponse: Response;
+          }
```

Comment:
This is actually = to general error. So we don't need to return this, but instead, convert this to a general error exception

Reply:
Done